src/example-types/facet.js

```diff
@@ -134,5 +138,58 @@ module.exports = {
           }),
         options
       ),
-    })),
+    }))
+    // results.options.name is ObjectId which need to stringify to get the correct missedValues
+    // stringify values to avoid the bug when values are numeric values
+    let missedValues = _.difference(
+      _.map(_.toString, values),
+      _.map(({ name }) => _.toString(name), results.options)
+    )
+
+    let getMissedValues = (node, missedValues) =>
+      node.isMongoId ? _.map(ObjectID, missedValues) : missedValues
+
+    let matchMissedValues = (node, missedValues) => ({
+      $match: { _id: { $in: getMissedValues(node, missedValues) } },
+    })
+
+    if (!_.isEmpty(missedValues)) {
+      let searchMissedValues = await search(
+        _.compact([matchMissedValues(node, missedValues)])
+      )
+
+      let stillMissingValues = _.difference(
+        missedValues,
+        _.map(({ _id }) => _.toString(_id), searchMissedValues)
+      )
+      let stillresult
+      if (stillMissingValues) {
+        console.log(matchMissedValues(node, stillMissingValues))
+        stillMissQuery = _.compact([
+          'stillMissing',
+          matchMissedValues(node, stillMissingValues),
+          { $group: { _id: `$${node.field}` } },
+          ...lookupLabel(node),
+          _.get('label.fields', node) && projectStageFromLabelFields(node),
+        ])
+      }
+      console.log(stillMissingValues)
+      console.log('stillresult')
+      console.log(stillresult[1]['$match'])
+      console.log(await search(stillMissQuery))
```

Comment:
@daedalus28 when there are stillMissingValues I supposed to get search results from this line. however, I always get []

Reply:
@chris110408 line 169 above -- the first element of a mongo aggregation can't be a string

Reply:
@doug thanks, it was the first approach, I had a checker on index request when the first element is 'stillmissing' the first match filter will be deleted and I use _.drop(1,aggs) delete the first element.  @daedalus28 provides me a better solution to use config.getProvider to run the search. the PR has been updated.